runtime/lua/vim/lsp.lua

```diff
@@ -1348,6 +1349,8 @@ function lsp.buf_detach_client(bufnr, client_id)
   local namespace = vim.lsp.diagnostic.get_namespace(client_id)
   vim.diagnostic.reset(namespace, bufnr)
 
+  nvim_exec_autocmds("LspDetached", { pattern = client.name, modeline = false })
```

Comment:
I think `LspDetached` also needs to be somewhere in `client.stop`?

Reply:
Added it to the `on_exit` handler (which is called after `client.stop` so should handle this case too). Thanks.